include/circt/Dialect/RTL/Combinatorial.td

```diff
@@ -71,6 +71,18 @@ class UTBinRTLOp<string mnemonic, list<OpTrait> traits = []> :
   }];
 }
 
+// Binary operator with uniform input types and i1 output type.
+class CmpBinRTLOp<string mnemonic, list<OpTrait> traits = []> :
+      BinRTLOp<mnemonic,
+               !listconcat(traits, [SameTypeOperands])> {
+  let arguments = (ins AnySignlessInteger:$lhs, AnySignlessInteger:$rhs);
+  let results = (outs I1:$result);
+
+  let assemblyFormat = [{
+    $lhs `,` $rhs  attr-dict `:` functional-type(operands, $result)
```

Comment:
I think you can simplify the syntax a bit by just requiring `type($lhs)` after the colon.  The rest of the types are unambiguously derivable from that.

Reply:
Done